File: new/atoms/cooldown-for-fails.test.mjs
import { cooldownForFails as __pure_cooldownForFails } from './cooldown-for-fails.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הקשיחים; בדיקה לא מייבאת אטום-שכן)
const __d_cooldown_for_fails_T = {
  k1: 30000,
  k2: 15000,
  k3: 5000,
};
const cooldownForFails = (...a) => __pure_cooldownForFails(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), __d_cooldown_for_fails_T);
const C = [[0, 0], [2, 0], [3, 5000], [4, 15000], [5, 30000], [7, 30000]];
let f = 0;
for (const [a, w] of C) { const g = cooldownForFails(a); if (g !== w) { console.error(`✗ ${a} ⇒ ${g} ≠ ${w}`); f = 1; } }
if (f) process.exit(1);
console.log('✓ cooldown-for-fails: 6 דוגמאות-חוזה — ירוק');

File: new/dart-maor/cooldown-for-fails_test.dart
// רתמת-זהב · cooldown-for-fails — assert-ים = דוגמאות-החוזה של בדיקת-ה-JS (זהות).
// אם עובר: Dart ≡ JS.
import 'cooldown-for-fails.dart';

void main() {
  const cases = <List<int>>[
    [0, 0],
    [2, 0],
    [3, 5000],
    [4, 15000],
    [5, 30000],
    [7, 30000],
  ];
  for (final c in cases) {
    final got = cooldownForFails(c[0]);
    assert(got == c[1], '✗ ${c[0]} ⇒ $got ≠ ${c[1]}');
  }
  print('✓ cooldown-for-fails (Dart): 6 דוגמאות-חוזה — ירוק');
}
